mimirtool: retain old default of blocks_storage.backend (#1626)
grafana/mimir: docs/sources/operators-guide/tools/mimirtool.md

| `-v`, `--verbose`    | If you set this flag, the CLI flags and YAML paths from the old configuration that do not exist in the new configuration are printed to `stderr`. This flag also prints default values that have changed between the old and the new configuration. |
| `--gem`              | If you set this flag, the tool will convert from Grafana Metrics Enterprise (GEM) v1.7.x to v2.0.0.                                                                                                                                                 |

> ##### A note on default values

Reviewer: ```suggestion
##### Changes to default values
```

> This command is designed to aid a migration from Cortex to Grafana Mimir. There are changes to the default values of

Reviewer: ```suggestion
The `mimirtool` command helps you migrate from Cortex to Grafana Mimir. There are changes to the default values of
```
I assume we are talking about the `mimirtool` here.

> some configuration parameters in Mimir v2.0.0 that we think you would not want to make as part of this migration. These

Reviewer: ```suggestion
some configuration parameters in Mimir v2.0.0 that you might not want to use as part of this migration:
```

> parameters are:

Reviewer: ```suggestion
```

- `blocks_storage.backend` - Cortex default is `s3`, Mimir default is `filesystem`
- `ruler_storage.backend` - Cortex default is `s3`, Mimir default is `filesystem`
- `alertmanager_storage.backend` - Cortex default is `s3`, Mimir default is `filesystem`
- `server.http_listen_port` - Cortex default is `80`, Mimir default is `8080`

> - (GEM only) `auth.type` - Cortex default is `trust`, Mimir default is `enterprise`

Reviewer 2: Nit: It's GEM default, not Mimir.

> This command will output the Cortex default value even when the configuration parameter is not explicitly set in the input
> configuration. If the Cortex default value is explicitly set in the input configuration, and you have provided
> the `--update-defaults` flag, the command will not update the value to the Mimir default.

Reviewer: ```suggestion
The `mimirtool` command outputs the Cortex default value even when the configuration parameter is not explicitly set in the input
configuration. If in the input configuration you explicitly set the Cortex default value, and you have provide
the `--update-defaults` flag, the `mimirtool` command will not update the value to the Mimir default.
```
Please vet for technical accuracy.

Author: changed in cecbdd27675f13e9375b09289347cb4186623398, can you take another look?
